llvm/lib/Analysis/StackSafetyAnalysis.cpp

```diff
@@ -273,6 +273,14 @@ ConstantRange StackSafetyLocalAnalysis::offsetFrom(Value *Addr, Value *Base) {
     return UnknownRange;
 
   auto *PtrTy = PointerType::getUnqual(SE.getContext());
+
+  // FIXME: Pass does not deal with pointers from different address spaces that
+  // don't have the same size.
+  if (DL.getTypeSizeInBits(Addr->getType()) != PointerSize ||
```

Comment:
Checking mismatched address space would also be better?

Reply:
This pass works fine for same size AS I think, because `getTruncateOrZeroExtend` will not try to do anything as it just checks the type size and not the types themselves.

I guess the easiest thing then is to just reject any AS != 0 explicitly

Reply:
> I guess the easiest thing then is to just reject any AS != 0 explicitly

Yes. This pass is just and optimization.